bin/weewx/reportengine.py

```diff
@@ -254,6 +254,113 @@ def _build_skin_dict(self, report):
         for scalar in self.config_dict['StdReport'].scalars:
             skin_dict[scalar] = self.config_dict['StdReport'][scalar]
 
+        #######################################################################
+        # internationalization support
+
+        # The key 'lang' defines a language code like 'en' or 'de'. It is
+        # used as a file name for a language file that is located in the
+        # 'lang' subdirectory of the skin directory. Like it is with the
+        # other path settings, a complete path overwrites the skin
+        # directory setting.
+
+        # If an localization file is defined, determine the path.
+        # As config_dict is not merged into skin_dict so far, we
+        # need to check both. config_dict (weewx.conf) has the final say,
+        # so check it first.
+        if 'lang' in self.config_dict['StdReport'][report]:
```

Comment:
Don't see why `lang` would not obey the same hierarchical principles as any other option. It shouldn't be necessary to do "look here, if not there."

Reply:
You say, `weewx.conf` has the last, final say. If I obey that rule and obey the hierarchical principles, I have to merge the language config into `skin_dict` ***before*** the data from weewx.conf are merged into `skin_dict`. At that moment, `skin_dict['lang']` has not the final value. So I have to take into account not only `skin_dict`, but also `self.config_dict`. 

Anyway. I found a way to replace those `if` - `elif` - `else` sequence by a simple single line of code.